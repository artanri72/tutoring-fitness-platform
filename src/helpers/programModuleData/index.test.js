// @flow

import { DateTime } from 'luxon';

import { programModuleDataFactory, programModuleDataToEvents } from '.';

test('programModuleDataToEvents() works', () => {
  const programModuleData = programModuleDataFactory.create(
    'generic',
    {
      scheduling: {
        startDay: 1,
        startTime: 9 * 60 * 60 * 1000,
        recurrenceFrequency: 'daily',
        recurOnDays: [],
        recurrenceInterval: 1,
        recurrenceEnd: 'never',
        recurrenceEndOnDay: 1,
        recurrenceEndAfterAmount: 1,
        configuration: {
          noNotifications: false,
          eventsViewableBeforeStartDate: false,
        },
      },
      action: {
        type: 'multiStep',
        steps: [
          {
            name: 'title',
            action: {
              type: 'blocks',
              blocks: [],
            },
          },
        ],
      },
    },
    { id: 'test', name: 'test' },
  );

  const options = {
    startDate: DateTime.fromISO('2019-06-10T21:21:10.219+03:00').setZone('utc'),
    endDate: DateTime.fromISO('2019-06-10T21:21:10.219+03:00').setZone('utc'),
  };
  const events = programModuleDataToEvents(options, programModuleData);

  expect(events).toMatchSnapshot();
});
