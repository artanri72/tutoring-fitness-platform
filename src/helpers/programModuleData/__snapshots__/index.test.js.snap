// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`programModuleDataToEvents() works 1`] = `
Array [
  Object {
    "end": "2019-06-10T09:00:00.000Z",
    "id": "test-0",
    "programModuleData": Object {
      "data": Object {
        "action": Object {
          "steps": Array [
            Object {
              "action": Object {
                "blocks": Array [],
                "type": "blocks",
              },
              "name": "title",
            },
          ],
          "type": "multiStep",
        },
        "scheduling": Object {
          "configuration": Object {
            "eventsViewableBeforeStartDate": false,
            "noNotifications": false,
          },
          "recurOnDays": Array [],
          "recurrenceEnd": "never",
          "recurrenceEndAfterAmount": 1,
          "recurrenceEndOnDay": 1,
          "recurrenceFrequency": "daily",
          "recurrenceInterval": 1,
          "startDay": 1,
          "startTime": 32400000,
        },
      },
      "id": "test",
      "name": "test",
      "type": "generic",
    },
    "start": "2019-06-10T09:00:00.000Z",
    "title": "test",
  },
]
`;
